src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -140,8 +140,8 @@ public void start(final Map<String, String> props) {
 
                 // If the cursor is invalidated add the record and return calls
                 if (changeStreamDocument.getString("operationType", new BsonString("")).getValue().equalsIgnoreCase(INVALIDATE)) {
-                    LOGGER.info("Cursor has been invalidated, no further messages will be published");
-                    isRunning.set(false);
+                    LOGGER.info("Cursor has been invalidated.");
+                    cursor = null;
```

Comment:
Should we be setting the resume token here to use when creating the new cursor, or is that already taken care of?

Reply:
Its already saved as part of the sourceRecord metadata